Perfil de participante

JLeonN opened the pull request.

JLeonN committed:
Perfil de participantes

Estoy trabajando en la creación de un componente para el Perfil de participantes.

JLeonN committed:
Cambios en la estructura

Mejorando la estructura para empezar a trabajar en el componente de ListaTitulos.

JLeonN committed:
Merge branch 'main' into PerfilDeParticipante-

francosang commented:
@JLeonN cuando esto este listo para revisar, acordate de ponerme como revisor o avisarme por discord

JLeonN committed:
Lista de titulos

JLeonN committed:
Formulario para administrar titulos.

Bueno, el formulario ya quedó listo.
Ahora solo falta ajustar el funcionamiento de la API para que muestre únicamente a un participante.

Voy a empezar a trabajar en eso ahora mismo.

francosang commented on frontend/src/Pagina/InfoTonto/ListaParticipantes.jsx:
```diff
@@ -1,20 +1,21 @@
-import ListaTitulos from "./ListaTitulos";
-import { useNombres } from "../../API/Api"
+// import ListaTitulos from "../PerfilDeParticipante/ListaTitulos";
```
Borrar este comentario.

francosang commented on frontend/src/Pagina/InfoTonto/ListaParticipantes.jsx:
```diff
@@ -1,20 +1,21 @@
-              <ListaTitulos />
+              <Link to="/EstructuraDePerfil" className="nombre fw-bold">{user.name}</Link>
+              <div className="fondoNav">
+                {/* <ListaTitulos /> */}
```
Borrar este comentario.

francosang commented on frontend/src/Pagina/PerfilDeParticipante/EstructuraDePerfil.jsx:
```diff
@@ -0,0 +1,24 @@
+import ListaTitulos from "./ListaTitulos";
+
+const EstructuraDePerfil = () => {
+  const { users } = useNombres();
```
Aca estas mostrando los titulos de todos los participantes.

Esta pantalla iba a ser para mostrar a un unico participante, no?

Es decir, ibamos a tener una pantalla que mostrara un participante y sus titulos.

Al menos, eso fue lo que habías diseñado:

![perfil](https://github.com/The-Cowboys/the-cowboy-day/assets/25726643/71a48364-2a9b-4550-b7c2-26cc87ba423b)

JLeonN replied:
Sí, exacto, Franco.
Aún no he tenido la oportunidad de hacerlo.
Estoy en contacto con Pablo para que me brinde su ayuda, o tal vez tú puedas ayudarme en la mañana.
La idea es crear una pantalla exclusiva para cada participante.

francosang commented on frontend/src/Pagina/PerfilDeParticipante/EstructuraDePerfil.jsx:
```diff
@@ -0,0 +1,24 @@
+
+  return (
+    <div>
+      {users.map((user, index) => (
```
Si sacas este `users.map(...)` y dejas solo el HTML vas a tener una unica lista de titulos para el participante seleccionado.

francosang replied:
Para obtener al participante selecionado tenes que usar este hook:

```js
import { useLocation } from 'react-router-dom';

const EstructuraDePerfil = () => {
  const  { state } = useLocation();
  console.log(state?.user.name);
```

francosang replied:
Para enviar al participante seleccionado tenes que hacerlo desde `ListaParticipantes.jsx`:

```jsx
              <Link
                to="/EstructuraDePerfil"
                state={{ user: user }}
                className="nombre fw-bold"
              >
```

El truco esta en pasar al `user` en el atributo `state`.

francosang commented on frontend/src/Pagina/PerfilDeParticipante/ListaTitulos.jsx:
```diff
@@ -0,0 +1,66 @@
+    dispatch({
+      type: "add_task",
+      titel: inputRef.current.value,
+    });
```
Despues del dispatch, se podría borrar el valor del input para que se vaya el texto escrito:

```js
    inputRef.current.value = null;
```

JLeonN replied:
No tengo idea de cómo hacerlo.
Quizás sea una pavada y yo esté pensando en hacer tremenda locura. Pero mira, le dediqué como media hora a resolver eso y no pude.

francosang commented on frontend/src/Pagina/PerfilDeParticipante/ListaTitulos.jsx:
```diff
@@ -0,0 +1,66 @@
+            placeholder="Titulo nuevo"
+            ref={inputRef}
+          />
+          <input type="submit" value="Agreguar" className="btn btn-dark" />
```
Se escribe "Agregar" 🤣 
Sin la `u`.

JLeonN replied:
paa 
Gracias

francosang reviewed (commented):
@JLeonN te dejé unos comentarios, pero lo más importante es que no estas usando la API para guardar estos valores.
Hay que hacer estas cosas:
 - Al abrir la pantalla, hay que llamar a `GET /api/cowboys/<id-cowboy>/titulos` para obtener los titulos de los usuarios.
 - Al agregar un titulo, hay que llamar a `POST /api/cowboys/<id-cowboy>/titulos` para guardar al titulo.
 - Al borrar un titulo, hay que llamar a `DELETE /api/titulos/<id-titulo>` para borrar al titulo.

Avisame si no entendes algo que nos conectamos por discord y lo charlamos.

JLeonN committed:
Ortografia

JLeonN committed:
Merge branch 'main' into PerfilDeParticipante-

JLeonN committed:
1 usuario
